libs/agno/agno/tools/firecrawl.py

```diff
@@ -112,3 +117,24 @@ def map_website(self, url: str) -> str:
         """
         map_result = self.app.map_url(url)
         return json.dumps(map_result.model_dump(), cls=CustomJSONEncoder)
+
+    def search(self, query: str, limit: Optional[int] = None):
+        """Use this function to search for the web using Firecrawl.
+
+        Args:
+            query (str): The query to search for.
+            limit (int): The maximum number of results to return.
+        """
+        params: Dict[str, Any] = {}
+        if self.limit or limit:
+            params["limit"] = self.limit or limit
+        if self.formats:
+            params["scrape_options"] = ScrapeOptions(formats=self.formats)  # type: ignore
+        if self.search_params:
+            params.update(self.search_params)
+
+        search_result = self.app.search(query, **params)
+        if search_result.success:
+            return json.dumps(search_result.data, cls=CustomJSONEncoder)
+        else:
+            return "Error: " + search_result.error
```

Comment:
```suggestion
            return "Error searching with the Firecrawl tool: " + search_result.error
```

not sure if their error will be clear enough